File: src/gasing/division/division5.rs
use std::time::Instant;

fn main() {
    let a: i64 = 123456789;
    let b: i64 = 5;

    // GASING Division
    let start = Instant::now();
    let mut hasil: i64 = 0;
    for _ in 0..10000000 {
        hasil = a * 2 / 10;
    }
    let gasing_time = start.elapsed().as_secs_f64();
    println!("GASING Division 5: {} seconds", gasing_time);

    // Computer Division
    let start = Instant::now();
    for _ in 0..10000000 {
        hasil = a / b;
    }
    let computer_time = start.elapsed().as_secs_f64();
    println!("General Division 5: {} seconds", computer_time);

    println!("\nGASING is {} times faster", computer_time/gasing_time);
}

File: src/gasing/division/division5.js
const a = 123456789;
const b = 5;

// GASING Division
let start = performance.now();
let hasil;
for (let i = 0; i < 10000000; i++) {
    hasil = Math.floor((a * 2) / 10);
}
let gasing_time = (performance.now() - start) / 1000;
console.log("GASING Division 5:", gasing_time, "seconds");

// Computer Division
start = performance.now();
for (let i = 0; i < 10000000; i++) {
    hasil = Math.floor(a / b);
}
let computer_time = (performance.now() - start) / 1000;
console.log("General Division 5:", computer_time, "seconds");

console.log("\nGASING is", computer_time/gasing_time, "times faster");